pyteal/ast/abi/array.py

```diff
@@ -98,6 +97,10 @@ def __str__(self) -> str:
 # until something like https://github.com/python/mypy/issues/3345 is added, we can't make the size of the array a generic parameter
 class StaticArray(Array[T]):
     def __init__(self, valueType: T, length: int) -> None:
+        if length < 0:
```

Comment:
nit: by parsing https://github.com/algorand/go-algorand/blob/master/data/abi/abi_type.go#L122, it is not allowed to let array length == 0.

Reply:
I think it _should_ be allowed by the SDKs, since it is allowed by the [spec](https://github.com/algorandfoundation/ARCs/blob/main/ARCs/arc-0004.md#types) (says `N >= 0`), but it's not very important

Related issue: https://github.com/algorand/js-algorand-sdk/issues/497

Reply:
oof, though it is meaningless to let static array length to be 0, which is 0 byte, but it should be okay to be valid... does not feel harmed.